### PR TITLE
fix(email): Add support for `sendEmailIfUnverified`

### PR DESCRIPTION
**Pull request by vbudhram**
Adds `sendEmailIfUnverified` param to delegate sending emails to auth-server.

Supports https://github.com/mozilla/fxa-content-server/pull/4155

Connects to https://github.com/mozilla/fxa-content-server/issues/4105

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ notifications:
     on_failure: change
 
 node_js:
-  - "0.10"
+  - "4.5"
```
Comment (vbudhram) [inline .travis.yml]: @vladikoff curious if you have an idea what could be going on with failing tests. The server mock tests pass locally for me but fail on travis when using node 0.10. When upgrading to 4.5, they pass. Both server mock and local tests fail with either 0.10 or 4.5. 

Travis logs seem to point to auth-mailer build errors? https://travis-ci.org/mozilla/fxa-js-client/jobs/160703028#L704

Comment (shane-tomlinson) [inline .travis.yml]: @vbudhram - Anything to do with this comment?

> Starting with io.js 3 and Node.js 4, building native extensions requires C++11-compatible compiler, which seems unavailable on this VM. Please read https://docs.travis-ci.com/user/languages/javascript-with-nodejs#Node.js-v4-(or-io.js-v3)-compiler-requirements.

**Comment by shane-tomlinson** [comment]
@vbudhram - when running the tests locally using `npm run test-local`, the tests stall in `lib/verifyCode.js->#verifyEmail`

```
groupEnd: signIn
group: verifyCode
Waiting for email...
PASS: #verifyEmail (79ms)
Waiting for email...
Waiting for email...
Waiting for email...
Waiting for email...
Waiting for email...
```

**Comment by shane-tomlinson** [comment]
Oh, `#verifyEmail` is passing, it's the one below, `#verifyEmailCheckStatus` that appears to be stalling. Specifically - https://github.com/mozilla/fxa-js-client/blob/master/tests/lib/verifyCode.js#L77

**Comment by vbudhram** [comment]
@shane-tomlinson Fixed all tests from yesterday (needed to rebase from master) and didn't update email count. Currently, tests are red from sauce labs `Error: 20 Sep 15:25:21 - Error: failed to bring up tunnel VM.`, gonna retry in a few.

**Comment by shane-tomlinson** [comment]
the js-client local tests are timing out now. :(

group: verifyCode
Waiting for email...
>> PASS: #verifyEmail (10ms)
Waiting for email...
Waiting for email...

The tests eventually time out and report everything as A-OK.

**Comment by vbudhram** [comment]
@shane-tomlinson I think my understanding of how the js-client tests worked was a little off and it didn't help that timeouts made it seem like it was passing :(, either way I think I finally got everything in order. Once you give the thumbs, I can make release and update content-server.

Thanks for helping!

**Pull request merged**